app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,182 @@
+class OrdersController < ApplicationController
+  before_action :require_login, only: [:index, :pending, :cancelled, :paid, :completed, :show]
+  before_action :get_order, only: [:confirm, :show]
+  before_action :get_order_items, only: [:index, :status]
+  before_action :get_order_item_revenue, only: :show
+
+  def checkout
+    # get @current_order info from carts controller
+    @cart_items = session[:cart]
+    # @cart_items looks like { "1" => 2 }
+    # want instance_vars @p1, @p2, etc. set equal to the keys of the hash
+    @order = Order.new
+    @order_item = OrderItem.new
+    @products = []
+    if !@cart_items.nil?
```

Comment:
you can do `if @cart_items` here, try to avoid double negatives as much as possible as it is harder to understand the true/false case. Same applies for line 50 :smile: